test/cljam/io/bigwig_test.clj

```diff
@@ -0,0 +1,286 @@
+       (= (:cir-tree ah) (:cir-tree bh))
+       (let [eps 1e-4
+             x (:total-summary ah)
+             y (:total-summary bh)]
+         (and (= (:bases-covered x) (:bases-covered y))
+              (-> (- (:min-val x) (:min-val y)) Math/abs (< eps))
+              (-> (- (:max-val x) (:max-val y)) Math/abs (< eps))
+              (-> (- (:sum-data x) (:sum-data y)) Math/abs (< eps))
+              (-> (- (:sum-squared x) (:sum-squared y)) Math/abs (< eps))))))
+
+(defn- same-wig?
+  "Returns true if the given arguments are same wig formats and same values."
+  [as bs]
+  (->> (map (fn [a b]
+              (and (= (:track a) (:track b))
+                   (= (:chr a) (:chr b))
+                   (= (:start a) (:start b))
+                   (= (:end a) (:end b))
+                   (-> (- (:value a) (:value b)) Math/abs (< 1e-4))))
+            as bs)
+       (every? true?)))
+
+(defn- same-bedgraph?
+  "Returns true if the given arguments are same BedGraph values."
+  [as bs]
+  (->> (map (fn [a b]
+              (and (= (:track a) (:track b))
+                   (= (:chr a) (:chr b))
+                   (= (:start a) (:start b))
+                   (= (:end a) (:end b))
+                   (-> (- (:value a) (:value b)) Math/abs (< 1e-4))))
+            as bs)
+       (every? true?)))
+
+(deftest reader
+  (testing "make reader instance"
+    (with-open [rdr (bigwig/reader test-bigwig-variable-file)]
+      (is (instance? cljam.io.bigwig.BIGWIGReader rdr))))
+
+  (testing "throw Exception"
+    (is (thrown? Exception (bigwig/reader "test-resources/bigwig/not-found.bigWig"))))
+
+  (testing "read bigWig file"
+    (with-open [r (bigwig/reader test-bigwig-fixed-file)]
+      (is (and (same-bigwig-headers? (.headers r)
+                                     (:headers test-bigwig-fixed-fields))
+               (same-wig? (bigwig/read-tracks r)
+                          (:tracks test-bigwig-fixed-fields)))))
+    (with-open [r (bigwig/reader test-bigwig-variable-file)]
+      (is (and (same-bigwig-headers? (.headers r)
+                                     (:headers test-bigwig-variable-fields))
+               (same-wig? (bigwig/read-tracks r)
+                          (:tracks test-bigwig-variable-fields)))))
+    (with-open [r (bigwig/reader test-bigwig-bedgraph-file)]
+      (is (and (same-bigwig-headers? (.headers r)
+                                     (:headers test-bigwig-bedgraph-fields))
+               (same-bedgraph? (bigwig/read-tracks r)
+                               (:tracks test-bigwig-bedgraph-fields)))))
+    (with-open [r (bigwig/reader test-bigwig-non-leaf-blocks-file)]
+      (is (and (same-bigwig-headers? (.headers r)
+                                     (:headers test-non-leaf-blocks-fields))
+               (same-wig? (bigwig/read-tracks r)
+                          (:tracks test-non-leaf-blocks-fields)))))))
+
+(deftest source-type-test
+  (testing "reader"
+    (with-open [server (http-server)]
```

Comment:
`http-server` is not used in this test.

bigwig reader doesn't support a remote file because it uses `RandomAccessFile`, so you should add a test for that like `cljam.io.sequence-test/source-type-test`.

https://github.com/chrovis/cljam/blob/14a2af3e400fa7404f7ff2c3ce27484c7ea96671/test/cljam/io/sequence_test.clj#L323